client/v2/algod/tealCompile.go

```diff
@@ -36,6 +36,10 @@ func (s *TealCompile) Sourcemap(Sourcemap bool) *TealCompile {
 
 // Do performs the HTTP request
 func (s *TealCompile) Do(ctx context.Context, headers ...*common.Header) (response models.CompileResponse, err error) {
-	err = s.c.post(ctx, &response, "/v2/teal/compile", s.source, headers)
+	if s.p.Sourcemap {
```

Comment:
@barnjamin Leaving some thoughts on how to remedy the broken query string parameter handling.  Let me know what you think?  Happy to discuss live.

First - Thanks for identifying + providing a fix for the problem.

I think my objectives for the PR are:
* Make the change _generator_-compatible.  Re-running the generator should _not_ cause a diff.
* Avoid changes impacting all routes.

Given those objectives, I see 2 paths forward:
* The path shown in the PR - Augment query string by-hand.
* Add parallel methods only used by the TEAL compile route.  I'm imagining a parallel `submitFormRaw` + upstream methods to pass query string parameters as a dedicated method argument.

A few thoughts:
* Either route requires updates to https://github.com/algorand/generator/blob/master/go_templates/query.vm#L239-L243.
* Philosophically, I prefer adding parallel methods to isolate impact + start moving towards a more robust solution.  In practice, I see it'd be significant code duplication.  I'd like to sleep on it, but I'm leaning towards taking the PR as is + _generator_ template modifications.

I'll likely add a go-algorand-sdk ticket to suggest providing a 1st class repo-wide solution.

Reply:
I just tweaked this to be a little more `future-proof` by encoding the get parameters with the existing `go-querystring` dependency.  This pattern _should_ allow us to add further parameters as needed to the struct alone and have them encoded without touching the URL generation logic

Reply:
@barnjamin https://github.com/algorand/go-algorand-sdk/pull/335/commits/da90c91574cf8d33d15389db0a64315c7162daa4 works for me provided _generator_ can be modified.  I'll leave the thread open until a pass at _generator_ is made.

Reply:
https://github.com/algorand/generator/pull/37 PR'd

Reply:
@barnjamin Resolving - The _generator_ change looks good.  Additionally, I opened an issue to document the underlying gap:  https://github.com/algorand/generator/issues/38.